schemas/engagementSchema.js

```diff
@@ -6,10 +6,10 @@ export const engagementSchema = yup.object().shape({
   description: yup.string().required(),
   customer: yup.string().required(),
   SFDC: yup.string().url(),
-  SE: yup.string().matches(/^\d+$/).optional(),
-  POC_Engineer: yup.string().matches(/^\d+$/).optional(),
+  SE: yup.string().matches(/^\d+$/).required(),
+  POC_Engineer: yup.string().matches(/^\d+$/).required(),
   statusCode: yup.number().positive().integer().required(),
-  testPlanId: yup.string().optional(),
+  testPlanId: yup.string().required(),
```

Comment:
I'm not sure if this should be required. If you create a new engagement, a test plan is NOT automatically added thus there will not be a testPlanId.

Reply:
Fixed this in a new branch, will update this.